ts/Series/Waterfall/WaterfallSeries.ts

```diff
@@ -638,7 +638,10 @@ class WaterfallSeries extends ColumnSeries {
         this: WaterfallSeries
     ): SVGPath {
 
-        let data = this.data,
+        let data = this.data.filter((d): Boolean =>
+                // Skip points where Y is null (#18636)
+                d.y !== null
```

Comment:
We can do this safer by testing for `isNumber`. That will also rule out things like `undefined`, `NaN` and `Infinity`.

```ts
let data = this.data.filter(isNumber)
```